toolchain/parse/testdata/generics/interface/require_name.carbon

```diff
@@ -0,0 +1,41 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/parse/testdata/generics/interface/require_name.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/parse/testdata/generics/interface/require_name.carbon
+
+interface Foo {
+  fn Foo1();
+}
+
+interface Bar {
+  require i32 impls Foo;
+}
```

Comment:
Can you add tests of incorrect syntax? There are a lot of possible typos here, and we should make sure error handling is working.

Just as a few examples:

```
// Missing components.
require;
require impls Foo;
require i32;
require i32 impls;

// Extra expressions.
require i32 i32 impls Foo;
require i32 impls Foo Foo;
```

Plus, what happens when there's a missing semicolon?